ports/nrf/modules/machine/i2c.c

```diff
@@ -105,43 +108,52 @@ mp_obj_t machine_hard_i2c_make_new(const mp_obj_type_t *type, size_t n_args, siz
 
     // get static peripheral object
     int i2c_id = i2c_find(args[ARG_NEW_id].u_obj);
-    const machine_hard_i2c_obj_t *self = &machine_hard_i2c_obj[i2c_id];
+    machine_hard_i2c_obj_t *self = &machine_hard_i2c_obj[i2c_id];
 
     if (args[ARG_NEW_scl].u_obj != MP_OBJ_NULL) {
-        self->i2c->init.scl_pin = args[ARG_NEW_scl].u_obj;
+        const pin_obj_t * p_scl_obj = args[ARG_NEW_scl].u_obj;
+        self->config.scl = p_scl_obj->pin;
     } else {
         nlr_raise(mp_obj_new_exception_msg_varg(&mp_type_ValueError,
                   "I2C SCL Pin not set"));
     }
 
     if (args[ARG_NEW_sda].u_obj != MP_OBJ_NULL) {
-        self->i2c->init.sda_pin = args[ARG_NEW_sda].u_obj;
+        const pin_obj_t * p_sda_obj = args[ARG_NEW_sda].u_obj;
+        self->config.sda = p_sda_obj->pin;
     } else {
         nlr_raise(mp_obj_new_exception_msg_varg(&mp_type_ValueError,
                   "I2C SDA Pin not set"));
     }
 
-    self->i2c->init.freq = HAL_TWI_FREQ_100_Kbps;
+    self->config.frequency = NRF_TWI_FREQ_100K;
 
-    hal_twi_master_init(self->i2c->instance, &self->i2c->init);
+    // Set context to this object.
+    nrfx_twi_init(self->p_twi, &self->config, NULL, (void *)self);
 
     return MP_OBJ_FROM_PTR(self);
 }
 
-#include <stdio.h>
-
 int machine_hard_i2c_readfrom(mp_obj_base_t *self_in, uint16_t addr, uint8_t *dest, size_t len, bool stop) {
     machine_hard_i2c_obj_t *self = (machine_hard_i2c_obj_t *)self_in;
 
-    hal_twi_master_rx(self->i2c->instance, addr, len, dest, stop);
+    nrfx_err_t err_code = nrfx_twi_rx(self->p_twi, addr, dest, len);
+
+    if (err_code != NRFX_SUCCESS) {
+        return -err_code;
```

Comment:
`readfrom` and `writeto` should return a MP_Esomething error (for example MP_EIO).